config/config.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"strconv"
+
+	"github.com/kelseyhightower/envconfig"
+	"gopkg.in/yaml.v3"
+)
+
+// ReadConfig gets the GlobalConfig from a configFile (that is a path to the file).
+func ReadConfig(configFile string) (*Config, error) {
+	if len(configFile) == 0 {
+		fmt.Fprintln(os.Stderr, "No config file provided, configuration is reading from System environment")
+		return readConfigFromENV()
+	}
+	fmt.Fprintln(os.Stderr, "Configuration is reading from configuration file")
+	return readConfigFromYAML(configFile)
+}
+
+func readConfigFromYAML(configFile string) (*Config, error) {
+	b, err := ioutil.ReadFile(configFile)
+	if err != nil {
+		return nil, err
+	}
+	res := new(Config)
+	err = yaml.Unmarshal(b, res)
+	return res, err
+}
+
+func readConfigFromENV() (*Config, error) {
+	res := new(Config)
+	err := res.unmarshalENV()
+	return res, err
+}
+
+// LogFormat is the type used for describing the format of logs.
+type LogFormat string
+
+const (
+	// JSONFormat is used for JSON logs.
+	JSONFormat LogFormat = "json"
+	// TextFormat is used of structured text logs.
+	TextFormat LogFormat = "text"
+)
+
+var mapLogFormat = map[LogFormat]bool{ // nolint: gochecknoglobals
+	JSONFormat: true,
+	TextFormat: true,
+}
+
+// Config contains the configuration for a server.
+type Config struct {
+	LogFormat     LogFormat `yaml:"log_format"`
```

Comment:
As you maybe noticed, I removed the RPC config because it's actually a log format. So it is merged with the attribute `LogFormat`

Reply:
like that it is simplified a bit more the usage of the config and how it is used